R/calcRenShareTargets.R

```diff
@@ -0,0 +1,352 @@
+          "SE|Electricity|Hydro (EJ/yr)",
+          "SE|Electricity|Geothermal (EJ/yr)"
+        )
+
+        # variables to sum for total electricity
+        total.vars.toSum <- c(
+          "SE|Electricity|Coal (EJ/yr)",
+          "SE|Electricity|Gas (EJ/yr)",
+          "SE|Electricity|Oil (EJ/yr)",
+          "SE|Electricity|Nuclear (EJ/yr)",
+          share.vars.toSum
+        )
+      } else if (targetType == "SE|Electricity|Non-Biomass Renewable") {
+        # variables to sum for non-biomass renewable electricity
+        share.vars.toSum <- c(
+          "SE|Electricity|Solar (EJ/yr)",
+          "SE|Electricity|Wind (EJ/yr)",
+          "SE|Electricity|Hydro (EJ/yr)",
+          "SE|Electricity|Geothermal (EJ/yr)"
+        )
+
+        # variables to sum for total electricity
+        total.vars.toSum <- c(
+          "SE|Electricity|Coal (EJ/yr)",
+          "SE|Electricity|Gas (EJ/yr)",
+          "SE|Electricity|Oil (EJ/yr)",
+          "SE|Electricity|Nuclear (EJ/yr)",
+          "SE|Electricity|Biomass (EJ/yr)",
+          share.vars.toSum
+        )
+      } else if (targetType == "SE|Electricity|Non-Fossil") {
+        # variables to sum for non-fossil electricity
+        share.vars.toSum <- c(
+          "SE|Electricity|Solar (EJ/yr)",
+          "SE|Electricity|Wind (EJ/yr)",
+          "SE|Electricity|Hydro (EJ/yr)",
+          "SE|Electricity|Geothermal (EJ/yr)",
+          "SE|Electricity|Nuclear (EJ/yr)",
+          "SE|Electricity|Biomass (EJ/yr)"
+        )
+
+        # variables to sum for total electricity
+        total.vars.toSum <- c(
+          "SE|Electricity|Coal (EJ/yr)",
+          "SE|Electricity|Gas (EJ/yr)",
+          "SE|Electricity|Oil (EJ/yr)",
+          share.vars.toSum
+        )
+      }
+
+      xHistShareVar <- dimSums(IEA[, , share.vars.toSum], dim = 3)
+      xHistTotalVar <- dimSums(IEA[, , total.vars.toSum], dim = 3)
+
+      if (AvgSeveralYears) {
+        # average over 2018-2022
+        AvgYears <- c("y2018", "y2019", "y2020", "y2021", "y2022")
+        xHistShareVar <- collapseDim(dimSums(xHistShareVar[, AvgYears, ], dim = 2) / length(AvgYears))
+        xHistTotalVar <- collapseDim(dimSums(xHistTotalVar[, AvgYears, ], dim = 2) / length(AvgYears))
+      } else {
+        # only get 2020 year
+        xHistShareVar <- collapseDim(xHistShareVar[, "y2020", ])
+        xHistTotalVar <- collapseDim(xHistTotalVar[, "y2020", ])
+      }
+
+      # calculate renewable / non-fossil share in historical year by dividing share variable by total variable
+      xHistShare[, , targetType] <- xHistShareVar / xHistTotalVar
+      # for countries without data, set to share to 0
+      xHistShare[is.na(xHistShare)] <- 0
+      # get total
+      xHistTotal[, , targetType] <- xHistTotalVar
+    }
+
+    return(list(xHistShare,xHistTotal))
+  }
+
+
+  # function to extrapolate target years which are not REMIND timesteps to REMIND timesteps
+  extrapolateTargetYears <- function(x, xHist, targetYears) {
```

Comment:
if you decide to use nested functions, i recommend using a leading dot (e.g. `.extrapolateTargetYears` )  to make it clear to the reader that this is an internal function not used anywhere else